lib/reek/smells/smell_warning.rb

```diff
@@ -57,8 +57,8 @@ def matches_smell_details?(other_smell_details = {})
         common_parameters_equal?(other_smell_details)
       end
 
-      def matches?(klass, other_smell_details = {})
-        matches_smell_type?(klass) && matches_smell_details?(other_smell_details)
+      def matches?(klass, parameters = {}, vars = {})
+        smell_classes.include?(klass.to_s) && common_parameters_equal?(parameters) && vars_equal?(vars)
```

Comment:
That's is confusing now. The difference between "parameters" and "vars" is baroque.
I think we should rather beef up "matches_smell_details?" so that it deals with all "parameters" and "attributes" like this:

```Ruby
def matches?(klass, details = {})
  smell_classes.include?(klass.to_s) && matches_smell_details?(details)
end
```

with "details" looking like this:

```Ruby
{
  lines: [...], 
  message: ...,
  parameters: { name: ....}
}
```

And then in "matches_smell_details?" we compare those "details" in a similar, unified way similar to how we do it in "common_parameters_equal?" (which would then be obsolete).